### PR TITLE
Fix Backend docs search items

### LINKED ISSUE #100944: [docs bug] wrong indexing of class name
### 📚 The doc issue

Similar to #73631, there is document issue again for [torch.backends.cudnn](https://pytorch.org/docs/stable/backends.html#module-torch.backends.cudnn). It shall be `torch.backends.cudnn.benchmark`, but it is now `torch.backends.cudnn.torch.backends.cudnn.benchmark`.

image

This issue actually occurs for all attributes in [the backends page](https://pytorch.org/docs/stable/_sources/backends.rst.txt). I don't know if it happens for other pages.

### Suggest a potential alternative/fix

I'll link a PR soon.

cc @svekars @carljparker

### PR DESCRIPTION
**Pull request by ain-soph**
Fixes #100944

## New

image

image

## Old

image

image

### CODE REVIEW

**Comment by ain-soph** [comment]
There seems to be a warning exception raised in CI job, since I use `.. automodule:: torch.backends.cuda.matmul`.  
Is there any better solution for the search item?

```
WARNING: module torch.backends.cuda.matmul could not be imported: No module named 'torch.backends.cuda.matmul'
```

File: docs/source/backends.rst [hunk 1]
```diff
@@ -31,24 +31,31 @@ torch.backends.cuda
 
 .. autofunction::  torch.backends.cuda.is_built
 
-.. attribute::  torch.backends.cuda.matmul.allow_tf32
+.. automodule:: torch.backends.cuda.matmul
```
Comment (albanD) [inline docs/source/backends.rst:34]: I think the problem is that this is not a module but a custom object.
Maybe use autoclass? Or do only torch.backends.cuda ?

Comment (ain-soph) [reply docs/source/backends.rst]: I use `.. currentmodule` to replace `automodule` and it works now.

**Review by albanD — APPROVED** [review]
Perfect! Thanks!

**Comment by ain-soph** [comment]
@albanD Seems to fail with some unrelated reasons.

**Pull request closed without merging**